receiver/mongodbatlasreceiver/alerts.go

```diff
@@ -91,6 +117,49 @@ func newAlertsReceiver(logger *zap.Logger, cfg AlertConfig, consumer consumer.Lo
 }
 
 func (a alertsReceiver) Start(ctx context.Context, host component.Host) error {
+	switch a.mode {
+	case alertModeListen:
+		return a.startListening(ctx, host)
+	case alertModeRetrieval:
+		return a.startRetrieving(ctx, host)
+	default:
+		return fmt.Errorf("receiver attempted to start without a known mode: %s", a.mode)
+	}
+}
+
+func (a alertsReceiver) startRetrieving(ctx context.Context, host component.Host) error {
+	client, err := internal.NewMongoDBAtlasClient(a.publicKey, a.privateKey, a.retrySettings, a.logger)
+	if err != nil {
+		return err
+	}
+	a.client = client
+
+	// may want to consider using disk storage for tracking which alerts have been sent
+	// or may want to revisit how we get alerts
+	// this is a memory cache used to keep track of unupdated alerts
+	cache, err := buntdb.Open(":memory:")
+	if err != nil {
+		return fmt.Errorf("unable to initialize cache for retrieval client: %w", err)
+	}
+	a.cache = cache
+
+	var alerts []mongodbatlas.Alert
+	for _, p := range a.projects {
+		projectAlerts, err := a.client.GetAlerts(ctx, p.Name)
+		if err != nil {
+			return fmt.Errorf("unable to get alerts for project: %w", err)
+		}
+		alerts = append(alerts, projectAlerts...)
+	}
+	now := pcommon.NewTimestampFromTime(time.Now())
+	logs, err := a.convertAlerts(now, alerts)
+	if err != nil {
+		return err
+	}
+	return a.consumer.ConsumeLogs(ctx, logs)
+}
```

Comment:
This function looks like it should kick off a goroutine and scrape on a configured interval; As it is, I think this just scrapes once and never again?